src/clj_kondo/impl/analyzer.clj

```diff
@@ -2115,30 +2115,59 @@
-                             (:name ns))
-                           (namespace kw))
-            kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
-                       (str (name kw)))]
-        (when (or (zero? arg-count)
-                  (> arg-count 2))
-          (findings/reg-finding! ctx
-                                 (node->line (:filename ctx) expr :invalid-arity
-                                             (format "keyword :%s is called with %s args but expects 1 or 2"
-                                                     kw-str
-                                                     arg-count))))))))
+        config (:config ctx)
+        ns (:ns ctx)
+        ?resolved-ns (if namespaced?
+                       (if-let [kw-ns (namespace kw)]
+                         (or (get (:qualify-ns ns) (symbol kw-ns))
+                             ;; because we couldn't resolve the namespaced
+                             ;; keyword, we print it as is
+                             (str ":" (namespace kw)))
+                         ;; if the keyword is namespace, but there is no
+                         ;; namespace, it's the current ns
+                         (:name ns))
+                       (namespace kw))
+        kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
+                   (str (name kw)))]
+    (when (and (not (config/skip? config :invalid-arity callstack))
+               (or (zero? arg-count)
+                   (> arg-count 2)))
+      (findings/reg-finding! ctx
+                             (node->line (:filename ctx) expr :invalid-arity
+                                         (format "keyword :%s is called with %s args but expects 1 or 2"
+                                                 kw-str
+                                                 arg-count))))
+    ;; With one argument to the keyword call, we can check if the return is a map.
+    ;; Another linter warning we could create is to check if the 2-arity version
+    ;; will never return the default value.
+    (when (= arg-count 1)
+      (let [analyzed (first (analyze-children (update ctx :callstack conj [nil :list]) (rest (:children expr))))
+            ret (:ret analyzed)
+            call (:call ret)]
+        (cond
+          ;; Type from the configuration file.
+          (:tag ret)
+          (let [{:keys [:op :req]} (:tag ret)]
+            (when (= op :keys)
+              (get req kw)))
+
+          ;; FIXME: (ask in the PR): Do we already have a more elegant way of doing this?
+          ;; Inferred type.
+          call
+          (-> (deref (:namespaces ctx))
+              (get (:base-lang call))
+              (get (:lang call))
+              (get (:ns call))
+              :vars
+              (get (:name call))
+              :arities
+              (get (:arity call))
+              :ret
+              :val
+              (get kw)
+              :tag))))))
```

Comment:
Ugly, is there some better way to do it?

Reply:
Instead of looking this up at runtime, we postpone this until the `impl/linters.clj` phase, where we have a more complete picture of what is being called (e.g. from other namespaces).

Reply:
Keyword calls do not keep track of the argtypes atom, so I was not finding a good way to do it there, but I will try to go deeper into the code, thanks o/

Reply:
Hmm, good point, I don't know the right answer either until I take a deeper look.

Reply:
I will point to the locations of the `:resolver` key (the function which converts types dynamically) when you have some time o/

See https://github.com/clj-kondo/clj-kondo/pull/1541/files#diff-3f1a249587095acd795d57380e88790caca3cead9f2924b337637e2d905a3ff9R230-R247, which is used at https://github.com/clj-kondo/clj-kondo/pull/1541/files#diff-f0dd386221a7c2f1794f9ced61a738b313001f4fbc5336dc828b392e4d40e2bcR49.